.github/workflows/README.md

```diff
@@ -0,0 +1,30 @@
+# GitHub Actions Documentation
+
+## `file_naming.yaml`
+
+Runs on push, this action runs our perl script in `buildsystem/tools/file_naming_conventions.pl` to enforce certain name restrictions defined in our developer guidelines (P003). We use one action to checkout the code, then another to install perl before running the script.
+
+## `ornl_ascent_mirror.yaml`
+
+This pushes to a GitLab at ORNL and runs CI/CD on Ascent. This also can re-build modules for testing newer versions of ExaGO and it's dependencies without needing to monitor builds by hand.
+
+## `pnnl_mirror.yaml`
+
+Similar to `ornl_ascent_mirror.yaml`, this mirrors to PNNL GitLab, but also supports Incline, Decpeption and Newell.
+
+## `pre_commit.yaml`
+
+This enforces and runs pre-commit, and automatically commits fixes to any tests that it can. Noteably applies clang formatting, and cmake formatting most often, and requires developers to either install locally, or rebase to incorporate changes.
```

Comment:
```suggestion
This enforces and runs pre-commit, and automatically commits fixes to any tests that it can. 

Current checks include:
- clang formatting
- cmake formatting
- toml formatting
- yaml formatting
- end of file consistency
- forbidding new submodules

Usage of pre-commit either requires developers to either install locally or rebase to incorporate changes pushed by the bot to GitHub.
```